spec/unit/controllers/v3/apps_controller_spec.rb

```diff
@@ -1489,6 +1489,81 @@
     end
   end
 
+  describe '#clear_buildpack_cache' do
+    let(:app_model) { VCAP::CloudController::AppModel.make(droplet_guid: droplet.guid, desired_state: 'STARTED') }
+    let(:droplet) { VCAP::CloudController::DropletModel.make(state: VCAP::CloudController::DropletModel::STAGED_STATE) }
+    let(:space) { app_model.space }
+    let(:org) { space.organization }
+    let(:user) { VCAP::CloudController::User.make }
+
+    before do
+      set_current_user(user)
+      allow_user_read_access_for(user, spaces: [space])
+      allow_user_write_access(user, space:)
+      VCAP::CloudController::BuildpackLifecycleDataModel.make(app: app_model, buildpacks: nil, stack: VCAP::CloudController::Stack.default.name)
+    end
+
+    it 'returns a 202 and the app' do
+      post :clear_buildpack_cache, params: { guid: app_model.guid }, as: :json
+
+      expect(response).to have_http_status :accepted
+      expect(response['Location']).to eq("http://api2.vcap.me/v3/jobs/#{app_model.guid}")
+
+      execute_all_jobs(expected_successes: 1, expected_failures: 0)
+    end
+
+    context 'when the app does not exist' do
+      it 'raises an API 404 error' do
+        post :clear_buildpack_cache, params: { guid: 'blah' }, as: :json
+
+        expect(response).to have_http_status :not_found
+        expect(response.body).to include 'ResourceNotFound'
+      end
+    end
+
+    context 'permissions' do
+      context 'when the user does not have the write scope' do
+        before do
+          set_current_user(VCAP::CloudController::User.make, scopes: ['cloud_controller.read'])
+        end
+
+        it 'raises an ApiError with a 403 code' do
+          post :clear_buildpack_cache, params: { guid: app_model.guid }, as: :json
+
+          expect(response).to have_http_status :forbidden
+          expect(response.body).to include 'NotAuthorized'
+        end
+      end
+
+      context 'when the user cannot read the app' do
+        before do
+          disallow_user_read_access(user, space:)
+        end
+
+        it 'returns a 404 ResourceNotFound error' do
+          post :clear_buildpack_cache, params: { guid: app_model.guid }, as: :json
+
+          expect(response).to have_http_status :not_found
+          expect(response.body).to include 'ResourceNotFound'
+        end
+      end
+
+      context 'when the user can read but cannot write to the app' do
```

Comment:
Isn't there a way to test the CF roles directly?
SpaceDeveloper+SpaceSupport can clear the cache. Other roles (or no roles) can't.
Would be a perfect documentation as well for the intended authorisation.

Disclaimer: I'm not an expert in the capi unit tests. Maybe this is the standard way how to test it and then it is good.

Reply:
I have based these tests off the `start`, `stop` and `restart` method tests. So tried to keep them consistent across these controller actions.

Reply:
Something like this you @stephanme ?

https://github.com/cloudfoundry/cloud_controller_ng/blob/main/spec/unit/controllers/v3/apps_controller_spec.rb#L1307